src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,204 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private Node<T> newNode;
+    private Node<T> currentNode;
+    private Node<T> removed;
+    private int size;
+
     @Override
     public void add(T value) {
+        newNode = new Node<>(tail, value, null);
+
+        if (size == 0) {
+            head = tail = newNode;
+        } else {
+            tail.next = newNode;
+            newNode.prev = tail;
+            tail = newNode;
+        }
+
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexExclusive(index);
+
+        newNode = new Node<>(value);
+
+        if (head == null) {
+            head = tail = newNode;
+        } else if (index == 0) {
+            newNode.next = head;
+            head.prev = newNode;
+            head = newNode;
+        } else if (index == size) {
+            tail.next = newNode;
+            newNode.prev = tail;
+            tail = newNode;
```

Comment:
You have same logic like in add(value) method, you can combine logic and create private method